EasyPost.Tests/OrderTest.cs

```diff
@@ -1,116 +1,120 @@
+using Xunit;
 
 namespace EasyPost.Tests
 {
-    [TestClass]
-    public class OrderTest
+    public class OrderTest : UnitTest
     {
-        private TestUtils.VCR _vcr;
-
-        [TestInitialize]
-        public void Initialize()
+        public OrderTest() : base("order")
         {
-            _vcr = new TestUtils.VCR("order");
         }
 
-        private static async Task<Order> CreateBasicOrder()
-        {
-            return await Order.Create(Fixture.BasicOrder);
-        }
+        #region CRUD Operations
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Create]
         public async Task TestCreate()
         {
-            _vcr.SetUpTest("create");
-
-            Order order = await CreateBasicOrder();
-
-            Assert.IsInstanceOfType(order, typeof(Order));
-            Assert.IsTrue(order.id.StartsWith("order_"));
-            Assert.IsNotNull(order.rates);
-        }
-
-        [TestMethod]
-        public async Task TestRetrieve()
-        {
-            _vcr.SetUpTest("retrieve");
+            UseVCR("create");
 
             Order order = await CreateBasicOrder();
 
-            Order retrievedOrder = await Order.Retrieve(order.id);
-
-            Assert.IsInstanceOfType(retrievedOrder, typeof(Order));
-            // Must compare IDs since other elements of objects may be different
-            Assert.AreEqual(order.id, retrievedOrder.id);
+            Assert.IsType<Order>(order);
+            Assert.StartsWith("order_", order.id);
+            Assert.NotNull(order.rates);
         }
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Read]
         public async Task TestGetRates()
         {
-            _vcr.SetUpTest("get_rates");
+            UseVCR("get_rates");
 
             Order order = await CreateBasicOrder();
 
-            await order.GetRates();
+            await order.GetRates(); // this does not return anything, does actually update in-place
 
             List<Rate> rates = order.rates;
 
-            Assert.IsNotNull(rates);
-            foreach (var rate in rates)
-            {
-                Assert.IsInstanceOfType(rate, typeof(Rate));
-            }
-        }
-
-        [TestMethod]
-        public async Task TestBuy()
-        {
-            _vcr.SetUpTest("buy");
-
-            Order order = await CreateBasicOrder();
-
-            await order.Buy(Fixture.Usps, Fixture.UspsService);
-
-            List<Shipment> shipments = order.shipments;
-
-            foreach (var shipment in shipments)
+            Assert.NotNull(rates);
+            foreach (Rate rate in rates)
             {
-                Assert.IsInstanceOfType(shipment, typeof(Shipment));
-                Assert.IsNotNull(shipment.postage_label);
+                Assert.IsType<Rate>(rate);
             }
         }
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Read] // not really a read operation, since the only HTTP calls we're making are create and update methods
```

Comment:
Can we describe the **why** behind this. Meaning, why is it set to `[CrudOperations.Read]` when it's not really a read?

Reply:
Because it seemed like the best category, even though it wasn't perfect. Open to suggestions.

Reply:
Right, but what's the significance of the decorator? Is it for the ordering? If so, a note about "using Read here as the most logical ordering for these functions, even though they are not Read CRUD operations" or something to that effect.

Reply:
Gotcha. Yes, the attributes were introduced to help with auto-ordering to preserve CRUD order. I will clarify with a comment.

Reply:
Resolved.